### ISSUE TITLE
Build system regression in 0.7.0

### ISSUE THREAD

**Issue by dbandstra** [ziglang/zig#7228]
Apologies for the vague issue title, I haven't figured out yet what is going on.

`build.zig`:
```zig
const std = @import("std");

pub fn build(b: *std.build.Builder) void {
    b.step("one", "Run example 'one'").dependOn(
        &b.addExecutable("one", "main.zig").run().step,
    );
    b.step("two", "Run example 'two'").dependOn(
        &b.addExecutable("two", "main.zig").run().step,
    );
}
```

`main.zig`:
```zig
const std = @import("std");

pub fn main() void {
    std.debug.warn("hi\n", .{});
}
```

`zig build one` works, but if you then run `zig build two` you get this:
```zig
error: FileNotFound
two...The following command exited with error code 1:
/Users/dbandstra/zig-macos-x86_64-0.7.0/zig build-exe /Users/dbandstra/code/zang/repro/main.zig --cache-dir /User/dbandstra/code/zang/repro/zig-cache --name two --enable-cache
error: the following build command failed with exit code 1:
/Users/dbandstra/code/zang/repro/zig-cache/o/9ef725a5da104ba121029526507e6d3a/build /Users/dbandstra/zig-macos-x86_64-0.7.0/zig /Users/dbandstra/code/zang/repro /Users/dbandstra/code/zang/repro/zig-cache two
```

Same if you delete zig-cache and then `zig build two` (works) and then `zig build one` (doesn't work).

This worked in Zig 0.6.0.

**Comment by LemonBoy** [comment]
I believe this was fixed by #7271, I tested this example and zang and both of them work as expected.

**Comment by dbandstra** [comment]
Yes, thank you
